python/ray/serve/controller.py

```diff
@@ -420,13 +436,46 @@ def deploy_app(
                 "previous request."
             )
 
-        deployment_override_options = config.dict(
-            by_alias=True, exclude_unset=True
-        ).get("deployments", [])
-
         self.config_deployment_request_ref = run_graph.options(
             runtime_env=config.runtime_env
-        ).remote(config.import_path, config.runtime_env, deployment_override_options)
+        ).remote(
+            config.import_path,
+            config.runtime_env,
+            deployment_override_options,
+            version_dict,
+        )
+
+    def get_version_of_new_deployments(
+        self,
+        new_deployments: List[Dict],
+        old_deployments: List[Dict],
+        version_dict: Dict,
+    ):
+        """
+        Takes in a list of new deployment configs and old deployment configs.
+
+        Compares the configs, and if any field other than num_replicas, 
+        user_config, and autoscaling_config changed, bump the version for that 
+        deployment to a new random version.
+        """
+
+        new_deployments = {dep["name"]: dep for dep in new_deployments}
+        old_deployments = {dep["name"]: dep for dep in old_deployments}
+
+        def exclude_keys(dict):
```

Comment:
It's not clear right now that `exclude_keys` removes the keys that trigger only lightweight updates. Can we rename this function and the `excluded_keys` attribute to make that clearer? For example, we can call these keys `config_options` and mention in a comment that changing any other option constitutes a code change, which requires a new version.